### PR TITLE
Tweak message layout.

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m [hunk 1]
```diff
@@ -286,7 +283,21 @@ - (CGFloat)footerHeight
...
+- (CGFloat)footerHeight
+{
+    if (!self.showFooter) {
+        return 0.f;
+    }
+
+    return ceil(MAX(kExpirationTimerViewSize, self.footerLabel.font.lineHeight));
+}
+
+- (CGFloat)footerVSpacing
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m]: Add 1pt spacing between message bubble and status message.

File: Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m [hunk 2]
```diff
@@ -188,6 +188,7 @@ - (void)createContents
         UIView *quoteStripView = [UIView containerView];
         quoteStripView.backgroundColor = self.highlightColor;
         quoteStripView.userInteractionEnabled = NO;
+        quoteStripView.layer.cornerRadius = self.quotedReplyStripeRounding;
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m]: rounded end of quote stripe.

File: Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m [hunk 3]
```diff
@@ -444,7 +444,12 @@ - (CGFloat)quotedReplyStripeThickness
 
 - (CGFloat)quotedReplyStripeVExtension
 {
-    return 8.f;
+    return 4.f;
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSQuotedMessageView.m]: reduce the height of the quote stripe

**Comment by charlesmchen-signal** [comment]
simulator screen shot - iphone x - 2018-04-10 at 10 29 59

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m [hunk 4]
```diff
@@ -313,11 +324,17 @@ - (void)updateFooter
         !attributedText) {
         self.footerLabel.hidden = YES;
         [self.viewConstraints addObjectsFromArray:@[
+            [self.footerView autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:self.messageBubbleView],
             [self.footerView autoSetDimension:ALDimensionHeight toSize:0],
         ]];
         return;
     }
 
+    [self.footerView autoPinEdge:ALEdgeTop
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m]: Doe we need to capture this in the self.viewConstraints so as not to redundantly constrain?

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m [hunk 5]
```diff
@@ -270,13 +269,11 @@ - (void)updateDateHeader
     }
 }
 
-- (CGFloat)footerHeight
+- (BOOL)showFooter
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m]: naming nit: `shouldShowFooter`?

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m (hunk 4 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m]: Good catch.

File: Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m (hunk 5 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m]: Done.

**Pull request merged**